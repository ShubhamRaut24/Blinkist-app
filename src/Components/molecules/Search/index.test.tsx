
import SearchComponent from '.';
import renderer from 'react-test-renderer';

const bookData = {
    "id": 0,
    "name": "Bring Your Human to Work",
    "url": "https://s3-alpha-sig.figma.com/img/7a8d/2547/2a61cffee838cf588c90ce1fcfaedae7?Expires=1643587200&Signature=ekBIi7OlQPO21GI~8fPDOuOQ4nlCLyPuEP43UqRCQWyy6xtlmyoBlWm8mFVqvI~nXLT-zqA7mD~HHYsjrKefAJwQXn0C8KZmkePdKiCIIU3R~uprBblEYnzMf9X4lqtwkeQ-StYxP7evEzqJVoF0ZiYbWsrCuU-r2Hh-BuBswc9CKNSXsr240OGQUTNCBkpXLalMv1G5B8uMcGigrQZG3wNsyLn3Lhc7jAgkefLSuLIONUtnaPkNvQ6DFsoSTv5sPChSDGMi1GApj6LUBbp3Jy66ps6TMDAGXXQmIjRkbPqzlVBZxgrnnZ0EBwu8iVeznR56CdqArTLCbgCDnMop3A__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
    "timeRead": "13-minute read",
    "writerName": "Erica Keswin",
    "synopics": "Beyond Entrepreneurship 2.0 (2020) updates Jim Collins and Bill Lazier’s essential 1992 business handbook, Beyond Entrepreneurship for the entrepreneurs, visionaries, and innovators of today. This new edition combines the timeless business advice and strategy of the original text, supplemented with cutting-edge insights and case studies pertinent to today’s business world.",
    "for": "One of the key learnings for entrepreneurs is the understanding of the reality about success and how extraordinary opportunities, instead of talent, pave the way to excellence.",
    "about_author": "From Jim Collins, the most influential business thinker of our era, comes an ambitious upgrade of his classic, Beyond Entrepreneurship, that includes all-new findings and world-changing insights.",
    "aim": "Turning Your Business into an Enduring Great Company"

}


test('renders Text passed from prop', () => {
    const component = renderer.create(
    <SearchComponent books={bookData}   />
    );
  
  expect(component).toMatchSnapshot();
});
