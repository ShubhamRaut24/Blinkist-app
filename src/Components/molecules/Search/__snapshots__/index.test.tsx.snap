// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Text passed from prop 1`] = `
<div
  className="MuiBox-root css-15xoerd"
>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1f01bjz-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-label="description"
      className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
      id="standard-adornment-amount"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyUp={[Function]}
      placeholder="Search by title or author"
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw08pq-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiBox-root css-542qn1"
  >
    <ul
      className="MuiList-root MuiList-padding css-1inb9y4-MuiList-root"
    />
  </div>
</div>
`;
